File: Week 2- Arrays/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{
    char text[1000];
    printf("Text: ");
    fgets(text, sizeof(text), stdin);
    int letters = count_letters(text);
    int words = count_words(text);
    int sent = count_sentences(text);

    float l = (float) letters / words * 100;
    float s = (float) sent / words * 100;

    float index = 0.0588 * l - 0.296 * s - 15.8;
    int index1 = round(index);

    if (index1 < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index1 >= 2 && index1 <= 15)
    {
        printf("Grade %d\n", index1);
    }
    else if (index1 >= 16)
    {
        printf("Grade 16+\n");
    }
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            count++;
            while (isspace(text[i]))
            {
                i++;
            }
        }
    }
    return count;
}

int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count++;
        }
    }
    return count;
}

File: Week 6 - Python/sentimental-readability/readability.py
from cs50 import get_string
import math


def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count


def count_words(text):
    count = 0
    in_word = False
    for char in text:
        if char.isspace():
            if in_word:
                count += 1
                in_word = False
        else:
            in_word = True
    if in_word:
        count += 1
    return count


def count_sentences(text):
    count = 0
    for char in text:
        if char in '.!?':
            count += 1
    return count


def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    l = (letters / words) * 100
    s = (sentences / words) * 100

    index = 0.0588 * l - 0.296 * s - 15.8
    index = round(index)

    if index < 1:
        print("Before Grade 1")
    elif 1 <= index < 16:
        print(f"Grade {index}")
    else:
        print("Grade 16+")


if __name__ == "__main__":
    main()
